File: C1.UWP.FlexGrid/VB/EMenus/CellFactories/CategoryCellFactory.vb
Imports C1.Xaml.FlexGrid

#Region "ClassCategoryCellFactory"
Class CategoryCellFactory
    Inherits CellFactory
#Region "PrivateVariables"
    Private imgBtn As Button
    Private categoryCtrl As CategoryCtrl
    Private category As Category
#End Region

#Region "OverrideMethods"
    'Override CreateCell
    Public Overrides Function CreateCell(grid As C1FlexGrid, cellType As CellType, rng As CellRange) As FrameworkElement
        category = TryCast(grid(rng.Row, rng.Column), Category)
        If category Is Nothing Then
            Return MyBase.CreateCell(grid, cellType, rng)
        End If
        categoryCtrl = New CategoryCtrl(category.ImageUri, category.Name)
        imgBtn = categoryCtrl.ImgButton
        'Implement related events
        AddHandler imgBtn.Click, Function(s, e)
                                     TryCast(Window.Current.Content, Frame).Navigate(GetType(MainPage), category)
                                     Window.Current.Activate()
                                     Return True
                                 End Function
        Return categoryCtrl
    End Function
#End Region
End Class
#End Region

File: C1.UWP.FlexGrid/CS/EMenus/CellFactories/CategoryCellFactory.cs
using C1.Xaml.FlexGrid;
using Grapecity.C1_EMenus.Controls;
using Grapecity.C1_EMenus.Pages;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Controls;
using Windows.UI.Xaml.Media;
using Windows.UI.Xaml.Media.Imaging;

namespace Grapecity.C1_EMenus.CellFactories
{
    #region ClassCategoryCellFactory
    class CategoryCellFactory : CellFactory
    {
        #region PrivateVariables
        private Button imgBtn;
        private CategoryCtrl categoryCtrl;
        private Category category;
        #endregion

        #region OverrideMethods
        //Override CreateCell
        public override FrameworkElement CreateCell(C1FlexGrid grid, CellType cellType, CellRange rng)
        {
            category = grid[rng.Row, rng.Column] as Category;
            if (category == null)
                return base.CreateCell(grid, cellType, rng);
            categoryCtrl = new CategoryCtrl(category.ImageUri,category.Name);
            imgBtn = categoryCtrl.ImgButton;
            //Implement related events
            imgBtn.Click += (s, e) =>
            {
                (Window.Current.Content as Frame).Navigate(typeof(MainPage), category);
                Window.Current.Activate();
            };
            return categoryCtrl;
        }
        #endregion
    }
    #endregion
}
